actions/setup-gap/action.yml

```diff
@@ -76,31 +110,126 @@ runs:
+          openssl genrsa -out /tmp/setup-gap/ca.key 4096
+          openssl req -x509 -new -nodes -key /tmp/setup-gap/ca.key -sha256 -days 1 -out /tmp/setup-gap/ca.crt -subj "/CN=My CA"
+        fi
+
+    - name: Generate and Sign Server Certificate (K8s only)
+      if: inputs.use-k8s == 'true'
+      shell: bash
+      run: |
+        mkdir -p /tmp/setup-gap
+
+        echo "::debug::Generating server key and certificate signing request (CSR)"
+        openssl genrsa -out /tmp/setup-gap/server.key 2048
+        openssl req -new -key /tmp/setup-gap/server.key -out /tmp/setup-gap/server.csr -subj "/CN=localhost" -addext "subjectAltName=DNS:localhost,IP:127.0.0.1"
+
+        echo "::debug:: Generating SAN extension file"
+        echo -e "subjectAltName=DNS:localhost,IP:127.0.0.1" > /tmp/setup-gap/san.ext
+
+        echo "::debug::Signing server certificate with CA"
+        openssl x509 -req -in /tmp/setup-gap/server.csr -CA /tmp/setup-gap/ca.crt -CAkey /tmp/setup-gap/ca.key -CAcreateserial -out /tmp/setup-gap/server.crt -days 1 -sha256 -extfile /tmp/setup-gap/san.ext
+
+        echo "::debug::Removing CSR and SAN extension files"
+        rm /tmp/setup-gap/server.csr /tmp/setup-gap/san.ext
+
+    - name: Setup Kubeconfig (K8s only)
+      if: inputs.use-k8s == 'true'
+      shell: bash
+      run: |
+        echo "::debug::Getting kubeconfig"
+        aws eks update-kubeconfig --name ${{ inputs.k8s-cluster-name }} --region ${{ inputs.k8s-cluster-aws-region || inputs.aws-region }}
+
+        # Get cluster arn for modifying kubeconfig
+        CLUSTER_ARN=$(aws eks describe-cluster --name ${{ inputs.k8s-cluster-name }} --region ${{ inputs.k8s-cluster-aws-region || inputs.aws-region }} --query "cluster.arn" --output text)
+        echo "::add-mask::${CLUSTER_ARN}"
+
+        echo "::debug::Setting up kubeconfig for localhost proxy"
+
+        # Set the certificate-authority to the certificate authority configured above
+        kubectl config set clusters.$CLUSTER_ARN.certificate-authority /tmp/setup-gap/ca.crt
+
+        # Remove certificate-authority-data populated from `aws eks update-kubeconfig`
+        kubectl config unset clusters.$CLUSTER_ARN.certificate-authority-data
+
+        # Update the server to use the proxy
+        kubectl config set clusters.$CLUSTER_ARN.server https://localhost:${{ inputs.proxy-port }}/primary
+
+    - name: Login to AWS ECR (public ecr only)
+      if: inputs.use-private-ecr-registry != 'true'
       uses: aws-actions/amazon-ecr-login@062b18b96a7aff071d4dc91bc00c4c1a7945b076 # v2.0.1
       env:
         AWS_REGION: us-east-1
       with:
         registry-type: public
 
-    - name: Run aws sig4 proxy container
+    - name: Run aws sig4 proxy container (public ecr only)
+      if: inputs.use-private-ecr-registry != 'true'
       shell: bash
       env:
         AWS_ACCESS_KEY_ID: ${{ env.AWS_ACCESS_KEY_ID }}
```

Comment:
where are these getting initially set?

Reply:
The top of the action assumes an aws role.